src/resources/startup.py

```diff
@@ -293,10 +297,45 @@ def Execute(self, request, context):
         return execute_response
 
 
+def watch_connections(address, event):
+    number_of_failures = 0
+    timeout_sec = 1
+
+    while True:
+        # If there are three failures, shutdown the gRPC server.
+        if number_of_failures == 3:
+            event.set()
+            break
+
+        channel = grpc.insecure_channel(address)
+        try:
+            channelz_stub = channelz_pb2_grpc.ChannelzStub(channel)
+            # Wait for the channel to be ready.
+            grpc.channel_ready_future(channel).result(timeout=timeout_sec)
+        except grpc.FutureTimeoutError:
+            number_of_failures += 1
+            continue
+        servers = channelz_stub.GetServers(
+            channelz_pb2.GetServersRequest(start_server_id=0))
+        sockets = channelz_stub.GetServerSockets(
+            channelz_pb2.GetServerSocketsRequest(
+                server_id=servers.server[0].ref.server_id, start_socket_id=0))
+
+        # There should be always more than one socket connected to the server
+        if len(sockets.socket_ref) == 1:
+            number_of_failures += 1
+        else:
+            number_of_failures = 0
+
+        # Sleep for 4 seconds before polling next time
+        time.sleep(4)
+
+
 if __name__ == "__main__":
```

Comment:
I thought there was also a case where if triton just dies then the python process can be left running (zombie). How does this fix that case?

Reply:
Mentioned in the comment above how this fixes it. https://github.com/triton-inference-server/python_backend/pull/10/files/bb86d2ba9ddeba7df1a2dc3b461f7b423addd65e#r496131640